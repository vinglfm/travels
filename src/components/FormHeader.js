import React, { PureComponent } from 'react';
import styles from './Auth/AuthModal.css';

export default class FormHeader extends PureComponent {
  render() {
    return (
      <div className={styles.modal__header}>
        <a onClick={this.props.onBack} className={styles.modal__back}><i className='fas fa-arrow-left'></i></a>
        <span>{this.props.title}</span>
        <hr className={styles.modal__divider}/>
      </div>      
    );
  }
}