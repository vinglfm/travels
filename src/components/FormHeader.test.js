import enzyme, {shallow} from 'enzyme';
import ReactAdapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new ReactAdapter() });

import React from 'react';
import FormHeader from './FormHeader';

describe('<FormHeader />', () => {
  let wrapper;
  
  beforeEach(() => {
    wrapper = shallow(<FormHeader/>);
  });

  it('renders form header pure component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
