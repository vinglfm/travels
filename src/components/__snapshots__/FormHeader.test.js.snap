// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormHeader /> renders form header pure component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormHeader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <a
          className={undefined}
          onClick={undefined}
>
          <i
                    className="fas fa-arrow-left"
          />
</a>,
        <span />,
        <hr
          className={undefined}
/>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="fas fa-arrow-left"
/>,
          "className": undefined,
          "onClick": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fas fa-arrow-left",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            className={undefined}
            onClick={undefined}
>
            <i
                        className="fas fa-arrow-left"
            />
</a>,
          <span />,
          <hr
            className={undefined}
/>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fas fa-arrow-left"
/>,
            "className": undefined,
            "onClick": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-arrow-left",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
